src/preamble.js

```diff
@@ -1863,33 +1845,10 @@ function integrateWasmJS() {
 
   // do-method functions
 
-#if BINARYEN_METHOD != 'native-wasm'
-  function doJustAsm(global, env, providedBuffer) {
-    // if no Module.asm, or it's the method handler helper (see below), then apply
-    // the asmjs
-    if (typeof Module['asm'] !== 'function' || Module['asm'] === methodHandler) {
-      if (!Module['asmPreload']) {
-        // you can load the .asm.js file before this, to avoid this sync xhr and eval
-        {{{ makeEval("eval(Module['read'](asmjsCodeFile));") }}} // set Module.asm
-      } else {
-        Module['asm'] = Module['asmPreload'];
-      }
-    }
-    if (typeof Module['asm'] !== 'function') {
-      err('asm evalling did not set the module properly');
-      return false;
-    }
-    return Module['asm'](global, env, providedBuffer);
-  }
-#endif // BINARYEN_METHOD != 'native-wasm'
-
   function doNativeWasm(global, env, providedBuffer) {
```

Comment:
Can we also remove `doNativeWasm` and roll into its caller (or rename it to reflect what it does rather than how it's different from the other methods)?

Reply:
Good idea, renaming to `createWasm` and improving the comment.